internal/controller/vrg_volrep.go

```diff
@@ -2407,6 +2412,17 @@ func (v *VRGInstance) processPVSecrets(pv *corev1.PersistentVolume) error {
 		pv.Spec.CSI.ControllerExpandSecretRef = secFromSC
 	}
 
+	// the value for provisionerSecretName and provisionerDeletionSecretName are allways same. so we check
+	// if provisionerSecretName name exists in storageClass and if exists we populate PV Annotation with
+	// provisionerDeletionSecretName with the values from  provisionerSecretName and its namespace values.
+	secFromSC, exists = secretsFromSC(sc.Parameters, provisionerSecretName, provisionerSecretNamespace)
+	if exists {
+		pv.Annotations[provisionerDeletionSecretName] = secFromSC.Name
```

Comment:
Use util.AddAnnotation to add annotations, which also helps deal with nil map to create one. the test failure is due to attempting to write to a nil map in this case. (here and in other lines that add the annotation).

Reply:
updated the code to make use of `AddAnnotation()`from utils.